joomla/joomla-cms
# [4.3] Add a CLI installation for Joomla

This PR adds a method to install Joomla via command line to simplify API tests and installations with one-click-setups. The new installation method can be triggered by running `php installation/joomla.php install` in the root of the Joomla folder. The different options are documented by running `php installation/joomla.php help install`.

All options can be given either by option directly on the command line or when running the install command interactively. If a default value is given for an option and that option is not defined, that default value is taken.

### Summary of Changes
This adds a new CLI entry file, similar to `cli/joomla.php` to the installation with all necessary steps to run the installation. The code uses as much of the current installation as possible and validates the options from the `setup.xml` file.

![joomla-cli-options](https://user-images.githubusercontent.com/313866/180768047-960b425a-d97f-484f-bbb8-a8a2ee0b60ef.png)

### Testing Instructions
1. Run `php installation/joomla.php install` in your installation and check that afterwards Joomla is properly installed.
2. Run `php installation/joomla.php install` with all necessary CLI options

If you are using a dev version (checked out the GIT version) the installation folder is preserved, otherwise the folder is deleted.

### Documentation Changes Required
The CLI installation is documented here: https://docs.joomla.org/J4.x:Joomla_CLI_Installation as well as in `installation/INSTALL`.

### Acknowledgement
I'd like to thank [Mittwald](https://www.mittwald.de/) for supporting this improvement.
